File: SMP/SMP.Web/Modules/Productos/TblproductosUber/TblproductosUberForm.cs
﻿using Serenity.ComponentModel;

namespace SMP.Productos.Forms;

[FormScript("Productos.TblproductosUber")]
[BasedOnRow(typeof(TblproductosUberRow), CheckNames = true)]
public class TblproductosUberForm
{
    public string StrDivision { get; set; }
    public string StrCategoria { get; set; }
    public string StrSubCat { get; set; }
    public string StrGrupo { get; set; }
    public string StrUpc { get; set; }
    public string StrMarca { get; set; }
    public string StrNombre { get; set; }
    public string StrFormato { get; set; }
    public string Strimagen { get; set; }
}

File: SMP/SMP.Web/Modules/ServerTypes/Productos/TblproductosUberForm.ts
import { StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface TblproductosUberForm {
    StrDivision: StringEditor;
    StrCategoria: StringEditor;
    StrSubCat: StringEditor;
    StrGrupo: StringEditor;
    StrUpc: StringEditor;
    StrMarca: StringEditor;
    StrNombre: StringEditor;
    StrFormato: StringEditor;
    Strimagen: StringEditor;
}

export class TblproductosUberForm extends PrefixedContext {
    static formKey = 'Productos.TblproductosUber';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!TblproductosUberForm.init)  {
            TblproductosUberForm.init = true;

            var w0 = StringEditor;

            initFormType(TblproductosUberForm, [
                'StrDivision', w0,
                'StrCategoria', w0,
                'StrSubCat', w0,
                'StrGrupo', w0,
                'StrUpc', w0,
                'StrMarca', w0,
                'StrNombre', w0,
                'StrFormato', w0,
                'Strimagen', w0
            ]);
        }
    }
}